sopel/builtins/xkcd.py

```diff
@@ -70,19 +62,20 @@ def searchxkcd_search(query):
         response = requests.post(SEARCHXKCD_API, params=parameters)
     except requests.exceptions.ConnectionError as e:
         LOGGER.debug("Unable to reach searchxkcd API: %s", e)
-        return None
+        raise SearchFailedError(str(e))
     except Exception as e:
         LOGGER.debug("Unexpected error calling searchxkcd API: %s", e)
-        return None
+        raise SearchXkcdError(str(e))
 
     try:
         hits = response.json()['results']['hits']
         if not hits:
-            return None
+            raise NoResultsError
         first = hits[0]['objectID']
     except (JSONDecodeError, LookupError):
-        LOGGER.debug("Data format from searchxkcd API could not be understood.")
-        return None
+        msg = "Data format from searchxkcd API could not be understood."
+        LOGGER.debug(msg)
```

Comment:
Huh, I would have expected a warning here

Reply:
To be fair I often err on the side of "don't spew stuff into logs if the user isn't actively debugging", but this one doesn't carry a big dump of stuff with it. What do you think of a warning with the user-friendly error message here and then a debug logging call with the API response?